packages/debugger/src/panels/variables/tree.tsx

```diff
@@ -257,35 +250,43 @@ const VariableComponent = (props: IVariableComponentProps): JSX.Element => {
       <span>: </span>
       <span style={styleType}>{convertType(variable)}</span>
       <span className="jp-DebuggerVariables-hspacer"></span>
-      {service.model.hasRichVariableRendering && (
-        <button
-          className="jp-DebuggerVariables-renderVariable"
-          disabled={
-            !commands.isEnabled(Debugger.CommandIDs.renderMimeVariable, {
-              name: variable.name,
-              variablesReference: variable.variablesReference
-            } as any)
-          }
-          onClick={e => {
-            e.stopPropagation();
-            onSelection(variable);
-            commands
-              .execute(Debugger.CommandIDs.renderMimeVariable, {
+      {service.model.hasRichVariableRendering &&
+        // Don't add rich display for special entries
+        // debugpy: https://github.com/microsoft/debugpy/blob/cf0d684566edc339545b161da7c3dfc48af7c7d5/src/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_utils.py#L359
+        ![
+          'special variables',
+          'protected variables',
+          'function variables',
+          'class variables'
+        ].includes(variable.name) && (
```

Comment:
Ultimately, would it make sense to move it to settings so that users can customise it, or should we create a new field to the `inspectVariables` response and move the responsibility of knowing which variables are scopes without mime rendering support? I am happy to open a new issue for that.

Reply:
From the user point of view, it would be better if it does not need to know. So I'm :+1: to get them from the DAP connector.